File: VB/SalaryReports/Reports/EmployeeReport.vb
﻿Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace SalaryReports
	Partial Public Class EmployeeReport
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub EmployeeReport_ParametersRequestSubmit(ByVal sender As Object, ByVal e As DevExpress.XtraReports.Parameters.ParametersRequestEventArgs) Handles MyBase.ParametersRequestSubmit
			Dim report As EmployeeReport = TryCast(sender, EmployeeReport)
			Dim year As Integer = Convert.ToInt32(report.Parameters("paramYear").Value)
			Dim month As Integer = Convert.ToInt32(report.Parameters("paramMonth").Value)

			report.Parameters("paramPaymentDate").Value = New DateTime(year, month, 1)
		End Sub

	End Class
End Namespace

File: CS/SalaryReports/Reports/EmployeeReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace SalaryReports
{
    public partial class EmployeeReport : DevExpress.XtraReports.UI.XtraReport
    {
        public EmployeeReport()
        {
            InitializeComponent();
        }

        private void EmployeeReport_ParametersRequestSubmit(object sender, DevExpress.XtraReports.Parameters.ParametersRequestEventArgs e)
        {
            EmployeeReport report = sender as EmployeeReport;
            int year = Convert.ToInt32(report.Parameters["paramYear"].Value);
            int month = Convert.ToInt32(report.Parameters["paramMonth"].Value);

            report.Parameters["paramPaymentDate"].Value = new DateTime(year, month, 1);
        }

    }
}
